From 39f740068ad21618f6f70b6e37855c61cadbd716 Mon Sep 17 00:00:00 2001
From: Michael J Rubinsky <mrubinsk@horde.org>
Date: Tue, 25 Sep 2018 20:21:56 -0400
Subject: [PATCH] Bug: 14857

Escape user-provided resource name when outputting. Prevents XSS vuln.
---
 js/kronolith.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/kronolith.js b/js/kronolith.js
index 60aeac8c..d3b19dbf 100644
--- a/js/kronolith.js
+++ b/js/kronolith.js
@@ -3322,7 +3322,7 @@ KronolithCore = {
         }
         $('kronolithCalendarresourcegroupmembers').update();
         $H(Kronolith.conf.calendars.resource).each(function(r) {
-            var o = new Element('option', { value: r.value.id }).update(r.value.name);
+            var o = new Element('option', { value: r.value.id }).update(r.value.name.escapeHTML());
             $('kronolithCalendarresourcegroupmembers').insert(o);
         });
     },
